### ISSUE TITLE
Use this with fork of rust toolchain?

### ISSUE THREAD

**Issue by damccull** [oxalica/rust-overlay#183]
Hello. I'm still pretty new to NixOS, Nix, Flakes, and all of it, but learning quickly.

I'd like to know if there's a way to supply a fork of the rust toolchain to this overlay by github reference so that I can ask it to use the esp-rs version of the rust compiler that supports xtensa. My current solution is to install the espup tool they provide and run its install feature upon entering my `nix develop` shell. But this, for some reason, redownloads the xtensa version of rust each time, which makes it impossible to use offline and annoying to wait on every time I want to enter the shell.

Ideally I'd just be able to provide an override to a commit from https://github.com/esp-rs/rust or similar and have it only build the source when necessary (during a manually specified commit update, for instance). Is this possible? This would install the xtensa rust version before the espup tool runs and allow it to detect the tool properly, avoid a re-download.

**Comment by damccull** [comment]
Ok, so the espup problem is due to a dynamic linker issue. But, I still would like to know if there's a way to specify a 3rd party fork of rust rather than the default one.

**Comment by oxalica** [comment]
Duplicate of #89
